cloud/wfm/src/executor/index.ts

```diff
@@ -0,0 +1,45 @@
+import * as Promise from 'bluebird';
+import { Process } from '../process';
+import { ProcessInstance } from '../process-instance';
+import { Task, TaskEventData, TaskStatus } from '../task';
+
+/**
+ * Executor engine for a Process
+ *
+ * Triggers instantiation and execution of a ProcessInstance
+ */
+export interface Executor {
+  process: Process;
+  instance?: ProcessInstance;
+  start(): void;
+}
+
+export interface InstanceRepository {
+  save(instance: ProcessInstance): Promise<ProcessInstance>;
+}
+
+class ExecutorImpl implements Executor {
+  public instance: ProcessInstance;
```

Comment:
think this one should have some other name; `processInstance = new Process()`, we can call it ProcessOrder; or ProcessSubmition, WDYT @nialldonnellyfh, @wtrocki?